test/negative-test.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+# this script runs a command and ensures that its exit code is not zero.
+
+if $@; then
```

Comment:
I get the idea behind this, but this gets us further from parity on Windows. Probably a no-op for this PR though

Reply:
We may want to have the negative test be a RunTests-level concept: In other words, like we specify stderr/predictions to check, the test should be able to specify if it is expecting a successful or a failure exit. With that said, this would not address the issue of existing shell script-based tests.